### PR TITLE
Add AMD_SNER 

### PR DESCRIPTION
**Pull request by youngyoonii**
Changes to be committed:
	new file:   mdistiller/distillers/AMD_SNER.py
	modified:   mdistiller/distillers/__init__.py
	modified:   mdistiller/distillers/_common.py
	modified:   mdistiller/engine/cfg.py
	modified:   mdistiller/models/imagenet/vit.py

### CODE REVIEW

File: mdistiller/models/imagenet/vit.py [hunk 1]
```diff
@@ -221,7 +221,8 @@ def vit_large_patch16_224(pretrained: bool = False, **kwargs) -> VisionTransform
     ImageNet-1k weights fine-tuned from in21k @ 224x224, source https://github.com/google-research/vision_transformer.
     """
     model_args = dict(patch_size=16, embed_dim=1024, depth=24, num_heads=16)
-    model = _create_vision_transformer('vit_large_patch16_224', pretrained=pretrained, **dict(model_args, **kwargs))
+    # model = _create_vision_transformer('vit_large_patch16_224', pretrained=pretrained, **dict(model_args, **kwargs))
```
Comment (nda111) [inline mdistiller/models/imagenet/vit.py]: 흔적기관 주석 지워주세요

File: mdistiller/distillers/AMD_SNER.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            distill_f_t = f_t_sner.clone()
+            proj_f_s = self.adapter_dict[f"adapter_{m_l:03d}"](f_s)
+            
+            # feature loss
+            loss_feat = loss_feat + F.mse_loss(proj_f_s, distill_f_t)
+            # info loss
+            cosine_sim = F.cosine_similarity(f_t_sner, f_t, dim=-1) 
+            loss_info = loss_info + (1.0 - cosine_sim).mean() 
+            # outlier loss
+            norms = patch_mod.norm(dim=-1)
```
Comment (nda111) [inline mdistiller/distillers/AMD_SNER.py]: `detach` 여기에서 해주세요

**Review by nda111 — CHANGES_REQUESTED** [review]
수정사항 둘만 반영해서 merge해주세요!
